rocAL/source/loaders/audio/audio_loader.cpp

```diff
@@ -199,8 +199,13 @@ AudioLoader::load_routine() {
     return LoaderModuleStatus::OK;
 }
 
+// Uses the same condition in audio_read_and_decode.cpp to check if enough samples are present
 bool AudioLoader::is_out_of_data() {
-    return (remaining_count() < 0);
+    return (remaining_count() < _batch_size);
```

Comment:
Theoretically, we will hit this condition if we are in last batch. So I think adding a new fuction called is_last_batch() is better and keep the original function for default behavior

Reply:
This is the same condition used in audio_read_and_decode/ image_read_and_decode files. And we stop the processing if number of samples is less than batch size.
Tested with PARTIAL, FILL and DROP, works as expected.